integrations/llama_index/src/ucai_llamaindex/toolkit.py

```diff
@@ -169,3 +178,46 @@ def tools(self) -> List[FunctionTool]:
             List[FunctionTool]: A list of tools available in the toolkit.
         """
         return list(self.tools_dict.values())
+
+
+def extract_properties(data: dict) -> dict:
```

Comment:
```suggestion
def extract_properties(data: dict[???, ???]) -> dict[???, ???]:
```

Can we add key and value types?